bundles/org.openhab.binding.nikobus/src/main/java/org/openhab/binding/nikobus/internal/handler/NikobusPcLinkHandler.java

```diff
@@ -329,6 +329,20 @@ private void refresh() {
         List<Thing> things = getThing().getThings().stream()
                 .filter(thing -> thing.getHandler() instanceof NikobusModuleHandler).collect(Collectors.toList());
 
+        // if there are command listeners then we need an open connection even if no modules exist
+        if (!commandListeners.isEmpty()) {
+            NikobusConnection connection = this.connection;
+            if (connection == null) {
+                return;
+            }
+            try {
+                connectIfNeeded(connection);
+            } catch (IOException e) {
+                connection.close();
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
```

Comment:
Should this function return at this point? Or should the handler still be refreshed even if the thing is offline?

Reply:
good point, continuing does not make much sense if the connection is not available.